pwndbg/heap/ptmalloc.py

```diff
@@ -965,12 +965,21 @@ def malloc_par(self):
     @pwndbg.lib.memoize.reset_on_objfile
     def malloc_alignment(self):
         """Corresponds to MALLOC_ALIGNMENT in glibc malloc.c"""
-        # i386 will override it to 16 when GLIBC version >= 2.26
-        # See https://elixir.bootlin.com/glibc/glibc-2.26/source/sysdeps/i386/malloc-alignment.h#L22
+        if pwndbg.gdblib.arch.current == "i386" and pwndbg.glibc.get_version() >= (2, 26):
+            # i386 will override it to 16 when GLIBC version >= 2.26
+            # See https://elixir.bootlin.com/glibc/glibc-2.26/source/sysdeps/i386/malloc-alignment.h#L22
+            return 16
+        # See https://elixir.bootlin.com/glibc/glibc-2.37/source/sysdeps/generic/malloc-alignment.h#L27
+        if hasattr(gdb.Type, "alignof"):
```

Comment:
This part is really dirty for me, we need to split the code just for old GDB versions again :/
Is there any other way we can get the `alignof()` result without GDB API?

Reply:
There is this special function/operator handled by GDB:
```
pwndbg> print _Alignof(long double)
$1 = 16
```
Can you check if it works for GDB older than 8.2?

I am afraid it may not be though, since [this thread aimed 8.2 milestone :/](https://gdb-prs.sourceware.narkive.com/KUwdtcc0/bug-cli-17095-new-print-alignof-type-or-expression). Here is the [patch](https://sourceware.org/legacy-ml/gdb-patches/2018-04/msg00469.html) when they added it.

Reply:
Btw also when we can't access something in Python in old GDB maybe we can fetch a C function from GDB binary and use it instead if this function is public/exported by GDB (so: not a `static` function...).

This can be done by loading the binary like you would load a shared object, with the `obj = ctypes.CDLL("")` and then using `obj.function(<args>)`, but I am not 100% sure if it would work here. From my quick tests it does not seem to work, e.g., for the `type_align` function :(.